custom-assembly-report-generator/generate_custom_assembly_report.py

```diff
@@ -0,0 +1,184 @@
+from get_assembly_report_url import *
+import data_ops
+import os
+import pandas
+import urllib
+
+
+def get_header_line_index(assembly_report_path):
+    try:
+        with open(assembly_report_path) as assembly_report_file_handle:
+            return next(line_index for line_index, line in enumerate(assembly_report_file_handle)
+                        if line.lower().startswith("# sequence-name") and "sequence-role" in line.lower())

```

Comment:
Wouldn't it be enough to check the first condition?

Reply:
All the comment lines start with # and to guard against one of those lines besides the column header starting with sequence name, I employed a second condition.